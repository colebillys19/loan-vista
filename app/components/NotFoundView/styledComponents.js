import styled from 'styled-components';

import { appColorA } from 'styleConstants';

export const NotFoundViewContainer = styled.div`
  align-items: center;
  display: inline-flex;
  flex-direction: column;
  margin: 5rem auto 0;
  width: 100%;
`;

export const WarningHeading = styled.h1`
  color: ${appColorA};
  font-size: 2.2rem;
  padding: 0;
`;
