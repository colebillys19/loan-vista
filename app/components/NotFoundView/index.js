/**
 * NotFoundView
 * @description ...
 */

import React from 'react';

import TabWrapper from 'components/_base-ui/TabWrapper';
import { WarningIcon } from 'images/icons';

import { NotFoundViewWrapper, WarningHeading } from './styledComponents';

const NotFoundView = () => (
  <TabWrapper>
    <NotFoundViewWrapper>
      <WarningIcon size="12rem" />
      <WarningHeading>page not found</WarningHeading>
    </NotFoundViewWrapper>
  </TabWrapper>
);

export default NotFoundView;
