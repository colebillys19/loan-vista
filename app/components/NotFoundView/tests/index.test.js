import React from 'react';
import { render } from 'react-testing-library';

import NotFoundView from '../index';

describe('<NotFoundView />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<NotFoundView />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<NotFoundView />);
    expect(firstChild).toMatchSnapshot();
  });
});
