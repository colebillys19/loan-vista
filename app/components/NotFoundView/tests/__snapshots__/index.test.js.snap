// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundView /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="misc-tab"
  class="TabWrapper-j2wsfb-0 beIMbY"
  id="misc-view"
>
  <div
    class="styledComponents__NotFoundViewWrapper-sc-1vxcvo3-0 jLgTwp"
  >
    NotFoundView
  </div>
</section>
`;
